src/proto/factories/layer_factory.cpp

```diff
@@ -229,26 +229,12 @@ factory_type const& get_layer_factory() noexcept
 
 template <typename TensorDataType, data_layout Layout, El::Device Device>
 std::unique_ptr<Layer>
-construct_layer_legacy(lbann_comm* comm,
-                       int training_dr_linearized_data_size,
-                       int num_parallel_readers,
-                       const lbann_data::Layer& proto_layer)
+construct_layer_legacy(lbann_comm* comm, const lbann_data::Layer& proto_layer)
```

Comment:
TODO for me: we can now get rid of this function and push the last builder into the factory :)